### DISCUSSION TITLE
Console does not pinpoint line number for warnings from within deck.gl render engine

### DISCUSSION THREAD

**Discussion by biniyam17** [visgl/deck.gl#6593] [General]
Hi - 

I am working with a huge data set (10s of thousands of data points) and using the Tile Layer to fetch and render with a GeoJson Layer. However, as part of my accessor call I am getting a warning from my state management library that I am accessing an index out of bounds. When I look at this in my console, I cannot pinpoint the line where warning is coming from. See screenshot. All I know is that in this case I am getting a `[mobx] Out of bounds read: 2` warning. 

It's possible this issue is caused by my state management library but my assumption is that since this warning is occurring within a Deck.GL layer, it's not as straightforward to see the exact line number. Is that the case? Does Deck.GL bubble up warnings just as they are received or is it modifying them in some way?

Thanks,

<img width="1173" alt="Screen Shot 2022-01-27 at 4 00 31 PM" src="https://user-images.githubusercontent.com/19158379/151463247-34e928d7-bf70-4fde-8a4e-ae12f625252b.png">

**Comment by ibgreen** [comment]
As far as I can tell, the top few stack frames are not from deck.gl, but rather from some library (`observablearray.ts`) that overrides array access, probably to add change detection and extra checks, and it is those overrides that throw the error, not deck.gl. All the deck.gl-related stack frames are already shown (in this case I conclude that you are using nebula.gl)

Per the stack trace, this function is receiving a bad parameter: https://github.com/uber/nebula.gl/blob/master/modules/overlays/src/html-overlay.tsx#L26.

Hope that helps a little

**Reply by biniyam17** [reply]
Hi Ib, this does help a lot. I ended up figuring out the bug. At that the time, it was hard to read the stack trace since it didn't include a line reference in my code but yes you are correct the function was receiving a bad parameter. Instead of a number[] it was an ObservableArray. Once I fixed the typing, the issue resolved itself. Sorry to have created this discussion for a non relevant question, and thanks for your help!
